src/hera/workflows/_mixins.py

```diff
@@ -274,6 +274,32 @@ class IOMixin(BaseMixin):
     inputs: InputsT = None
     outputs: OutputsT = None
 
+    def get_parameter(self, name: str) -> Parameter:
+        """Finds and returns the parameter with the supplied name.
```

Comment:
I was writing a comment about output parameters from DAGs to confirm the logic, and ended up finding it works really well! Leaving a code example here - maybe we can add it as an example alongside callable_dag.py? Maybe `callable_dag_with_outputs.py`?

```py
@script(constructor="runner")
def hello_with_output(name: str) -> Annotated[str, Parameter(name="output-message")]:
    return "Hello, {name}!".format(name=name)

with Workflow(
    generate_name="callable-dag-",
    entrypoint="calling-dag",
) as w:
    with DAG(
        name="my-dag-with-outputs",
        inputs=Parameter(name="my-dag-input"),
        outputs=Parameter(
            name="my-dag-output",
            value_from={"parameter": "{{hello.outputs.parameters.output-message}}"},  # Don't think we can improve this?
        ),
    ) as my_dag:
        # I believe this works within the f-string - get_parameter is called on the IOMixin to get the input parameter
        hello_with_output(name="hello", arguments={"name": f"hello {my_dag.get_parameter('my-dag-input')}"})

    with DAG(name="calling-dag") as d:
        t1 = my_dag(name="call-1", arguments={"my-dag-input": "call-1"})
        # Here, t1 is a Task from the called dag, so get_parameter is called on the Task to get the output parameter! 🚀 
        t2 = my_dag(name="call-2", arguments={"my-dag-input": t1.get_parameter("my-dag-output")})
        t1 >> t2
```